### ISSUE TITLE
Flaky io.questdb.cutlass.pgwire.PGJobContextTest.testUpdateAsync

### ISSUE THREAD

**Issue by bluestreak01** [questdb/questdb#2195]
### Describe the bug

Full log is attached below this snippet

```
2022-06-06T19:21:23.1209938Z 2022-06-06T19:21:23.119209Z I i.q.TestListener >>>> io.questdb.cutlass.pgwire.PGJobContextTest.testUpdateAsync
2022-06-06T19:21:23.2307818Z 2022-06-06T19:21:23.132101Z I i.q.c.AbstractCairoTest Starting test PGJobContextTest#testUpdateAsync
2022-06-06T19:21:23.2310652Z 2022-06-06T19:21:23.132672Z A pg-server listening on 0.0.0.0:8812 [fd=20 backlog=128]
2022-06-06T19:21:23.2312968Z 2022-06-06T19:21:23.160648Z I i.q.c.p.PGJobContextTest os scheduled [name=questdb-worker-707]
2022-06-06T19:21:23.2314818Z 2022-06-06T19:21:23.171849Z I i.q.c.p.PGJobContextTest started
2022-06-06T19:21:23.2316650Z 2022-06-06T19:21:23.172027Z I i.q.c.p.PGJobContextTest os scheduled [name=questdb-worker-708]
2022-06-06T19:21:23.2318771Z 2022-06-06T19:21:23.174141Z I i.q.c.p.PGJobContextTest os scheduled [name=questdb-worker-709]
2022-06-06T19:21:23.2320610Z 2022-06-06T19:21:23.178857Z I i.q.c.p.PGJobContextTest started
2022-06-06T19:21:23.2322414Z 2022-06-06T19:21:23.178918Z I i.q.c.p.PGJobContextTest os scheduled [name=questdb-worker-710]
2022-06-06T19:21:23.2324230Z 2022-06-06T19:21:23.179366Z I pg-server connected [ip=127.0.0.1, fd=23]
2022-06-06T19:21:23.2326099Z 2022-06-06T19:21:23.179605Z I i.q.c.p.PGConnectionContext protocol [major=3, minor=0]
2022-06-06T19:21:23.2327960Z 2022-06-06T19:21:23.179612Z I i.q.c.p.PGConnectionContext property [name=user, value=admin]
2022-06-06T19:21:23.2329853Z 2022-06-06T19:21:23.179615Z I i.q.c.p.PGConnectionContext property [name=database, value=qdb]
2022-06-06T19:21:23.2331760Z 2022-06-06T19:21:23.179616Z I i.q.c.p.PGConnectionContext property [name=client_encoding, value=UTF8]
2022-06-06T19:21:23.2333861Z 2022-06-06T19:21:23.179618Z I i.q.c.p.PGConnectionContext property [name=DateStyle, value=ISO]
2022-06-06T19:21:23.2335884Z 2022-06-06T19:21:23.179619Z I i.q.c.p.PGConnectionContext property [name=TimeZone, value=GMT]
2022-06-06T19:21:23.2337912Z 2022-06-06T19:21:23.179620Z I i.q.c.p.PGConnectionContext property [name=extra_float_digits, value=2]
2022-06-06T19:21:23.2342088Z 2022-06-06T19:21:23.180080Z I i.q.g.SqlCompiler batch [text=BEGIN]
2022-06-06T19:21:23.2346222Z 2022-06-06T19:21:23.180111Z I i.q.c.p.PGConnectionContext parse [fd=23, q=BEGIN]
2022-06-06T19:21:23.2347317Z 2022-06-06T19:21:23.180189Z I i.q.g.SqlCompiler batch [text=SET extra_float_digits = 3]
2022-06-06T19:21:23.2348378Z 2022-06-06T19:21:23.180202Z I i.q.c.p.PGConnectionContext parse [fd=23, q=SET extra_float_digits = 3]
2022-06-06T19:21:23.2349507Z 2022-06-06T19:21:23.180262Z I i.q.g.SqlCompiler batch [text=SET application_name = 'PostgreSQL JDBC Driver']
2022-06-06T19:21:23.2350685Z 2022-06-06T19:21:23.180280Z I i.q.c.p.PGConnectionContext parse [fd=23, q=SET application_name = 'PostgreSQL JDBC Driver']
2022-06-06T19:21:23.2352501Z 2022-06-06T19:21:23.180328Z I i.q.g.SqlCompiler batch [text=COMMIT]
2022-06-06T19:21:23.2354503Z 2022-06-06T19:21:23.180341Z I i.q.c.p.PGConnectionContext parse [fd=23, q=COMMIT]
2022-06-06T19:21:23.2356729Z 2022-06-06T19:21:23.180595Z I i.q.g.SqlCompiler batch [text=create table x (a long, b double, ts timestamp) timestamp(ts)]
2022-06-06T19:21:23.2359531Z 2022-06-06T19:21:23.180793Z I i.q.c.CairoEngine locked [table=`x`, thread=1363]
2022-06-06T19:21:23.2361679Z 2022-06-06T19:21:23.182010Z I i.q.c.p.WriterPool created [table=`x`, thread=1363]
2022-06-06T19:21:23.2363686Z 2022-06-06T19:21:23.182043Z I i.q.c.TableWriter open 'x'
2022-06-06T19:21:23.2365819Z 2022-06-06T19:21:23.182423Z I i.q.c.TableWriter switched partition [path='/tmp/junit13914628204950235942/dbRoot/x/default']
2022-06-06T19:21:23.2368188Z 2022-06-06T19:21:23.182494Z I i.q.c.CairoEngine unlocked [table=`x`]
2022-06-06T19:21:23.2370347Z 2022-06-06T19:21:23.182500Z I i.q.c.p.PGConnectionContext parse [fd=23, q=create table x (a long, b double, ts timestamp) timestamp(ts)]
2022-06-06T19:21:23.2373350Z 2022-06-06T19:21:23.182639Z I i.q.g.SqlCompiler batch [text=insert into x values (1, 2.0, '2020-06-01T00:00:02'::timestamp),(2, 2.6, '2020-06-01T00:00:06'::timestamp),(5, 3.0, '2020-06-01T00:00:12'::timestamp)]
2022-06-06T19:21:23.2376772Z 2022-06-06T19:21:23.182780Z I i.q.c.p.ReaderPool open 'x' [at=0:0]
2022-06-06T19:21:23.2379425Z 2022-06-06T19:21:23.183651Z I i.q.c.p.PGConnectionContext parse [fd=23, q=insert into x values (1, 2.0, '2020-06-01T00:00:02'::timestamp),(2, 2.6, '2020-06-01T00:00:06'::timestamp),(5, 3.0, '2020-06-01T00:00:12'::timestamp)]
2022-06-06T19:21:23.2399418Z 2022-06-06T19:21:23.183669Z I i.q.c.p.WriterPool >> [table=`x`, thread=1363]
2022-06-06T19:21:23.2400883Z 2022-06-06T19:21:23.183950Z I i.q.c.p.WriterPool << [table=`x`, thread=1363]
2022-06-06T19:21:23.2401819Z 2022-06-06T19:21:23.184039Z I i.q.c.p.WriterPool >> [table=`x`, thread=1361]
2022-06-06T19:21:23.2402774Z 2022-06-06T19:21:23.196763Z I i.q.g.SqlCompiler batch [text=update x set a=9 where b>2.5]
2022-06-06T19:21:23.2403792Z 2022-06-06T19:21:23.201112Z I i.q.g.SqlCodeGenerator JIT enabled for (sub)query [tableName=x, fd=23]
2022-06-06T19:21:23.2404840Z 2022-06-06T19:21:23.201252Z I i.q.c.p.PGConnectionContext parse [fd=23, q=update x set a=9 where b>2.5]
2022-06-06T19:21:23.2405883Z 2022-06-06T19:21:23.201275Z I i.q.c.p.WriterPool busy [table=`x`, owner=1361, thread=1362, reason=test lock]
2022-06-06T19:21:23.2407029Z 2022-06-06T19:21:23.201341Z I i.q.g.e.o.OperationFutureImpl published ASYNC writer command [name=UPDATE,tableName=x,instance=3]
2022-06-06T19:21:23.2408207Z 2022-06-06T19:21:23.201377Z I i.q.c.TableWriter received async cmd [type=3, tableName=x, tableId=1, correlationId=3, cursor=0]
2022-06-06T19:21:23.2409240Z 2022-06-06T19:21:23.201399Z I i.q.g.UpdateOperator updating [table=x instance=3]
2022-06-06T19:21:23.2410531Z 2022-06-06T19:21:23.201487Z I i.q.c.TableReader open partition /tmp/junit13914628204950235942/dbRoot/x/default [rowCount=3, partitionNameTxn=-1, transientRowCount=3, partitionIndex=0, partitionCount=1]
2022-06-06T19:21:23.2412176Z 2022-06-06T19:21:23.201620Z I i.q.g.UpdateOperator updating partition [partitionIndex=-1, rowPartitionIndex=0, rowPartitionTs=1970-01-01T00:00:00.000000Z, affectedColumnCount=1, prevRow=0, minRow=-1]
2022-06-06T19:21:23.2413594Z 2022-06-06T19:21:23.202142Z I i.q.c.TableWriter switched partition [path='/tmp/junit13914628204950235942/dbRoot/x/default']
2022-06-06T19:21:23.2414976Z 2022-06-06T19:21:23.202229Z I i.q.g.UpdateOperator columns purged locally [table=x, column=a, newColumnVersion=1]
2022-06-06T19:21:23.2416065Z 2022-06-06T19:21:23.202231Z I i.q.g.UpdateOperator update finished [table=x, instance=3, updated=2, txn=2]
2022-06-06T19:21:23.2417224Z 2022-06-06T19:21:23.202248Z I i.q.c.TableWriter published async command complete event [type=3,tableName=x,tableId=1,correlationId=3]
2022-06-06T19:21:23.2418375Z 2022-06-06T19:21:23.202859Z I i.q.g.e.o.OperationFutureImpl writer command QUERY_STARTED response received [instance=3]
2022-06-06T19:21:23.2419469Z 2022-06-06T19:21:23.202862Z I i.q.g.e.o.OperationFutureImpl writer command response received [instance=3]
2022-06-06T19:21:23.2420425Z 2022-06-06T19:21:23.203018Z I i.q.c.p.WriterPool << [table=`x`, thread=1361]
2022-06-06T19:21:23.2421282Z 2022-06-06T19:21:23.203343Z I i.q.g.SqlCompiler batch [text=x]
2022-06-06T19:21:23.2422724Z 2022-06-06T19:21:23.203503Z I i.q.c.TableReader open partition /tmp/junit13914628204950235942/dbRoot/x/default [rowCount=3, partitionNameTxn=-1, transientRowCount=3, partitionIndex=0, partitionCount=1]
2022-06-06T19:21:23.2424090Z 2022-06-06T19:21:23.203760Z I i.q.g.SqlCompiler plan [q=`select-choose a, b, ts from (select [a, b, ts] from x timestamp (ts))`, fd=23]
2022-06-06T19:21:23.2425124Z 2022-06-06T19:21:23.203871Z I i.q.c.p.PGConnectionContext parse [fd=23, q=x]
2022-06-06T19:21:23.2425985Z 2022-06-06T19:21:23.205751Z I i.q.c.p.PGWireServer closed
2022-06-06T19:21:23.2426843Z 2022-06-06T19:21:23.205767Z I i.q.c.p.PGJobContextTest cleaned [worker=1]
2022-06-06T19:21:23.2427676Z 2022-06-06T19:21:23.206899Z I i.q.c.p.PGWireServer closed
2022-06-06T19:21:23.2428533Z 2022-06-06T19:21:23.207062Z I i.q.c.p.PGJobContextTest cleaned [worker=0]
2022-06-06T19:21:23.2429480Z 2022-06-06T19:21:23.207079Z I pg-server disconnected [ip=127.0.0.1, fd=23, src=shutdown]
2022-06-06T19:21:23.2430327Z 2022-06-06T19:21:23.207148Z I pg-server closed
2022-06-06T19:21:23.2431122Z 2022-06-06T19:21:23.207363Z I i.q.c.TableWriter closed 'x'
2022-06-06T19:21:23.2432026Z 2022-06-06T19:21:23.207366Z I i.q.c.p.WriterPool closed [table=`x`, reason=IDLE, by=1361]
2022-06-06T19:21:23.2432993Z 2022-06-06T19:21:23.207463Z I i.q.c.p.ReaderPool closed 'x' [at=0:0, reason=IDLE]
2022-06-06T19:21:23.2434141Z 2022-06-06T19:21:23.218591Z I i.q.c.AbstractCairoTest Tearing down test PGJobContextTest#testUpdateAsync
2022-06-06T19:21:23.2622560Z 2022-06-06T19:21:23.244063Z E i.q.TestListener ***** Test Failed *****io.questdb.cutlass.pgwire.PGJobContextTest.testUpdateAsync : 
2022-06-06T19:21:23.2623775Z java.lang.AssertionError: Memory usage by tag: NATIVE_JIT, difference: 4096 expected:<8192> but was:<12288>
2022-06-06T19:21:23.2624465Z 	at org.junit.Assert.fail(Assert.java:89)
2022-06-06T19:21:23.2625035Z 	at org.junit.Assert.failNotEquals(Assert.java:835)
2022-06-06T19:21:23.2625629Z 	at org.junit.Assert.assertEquals(Assert.java:647)
2022-06-06T19:21:23.2626278Z 	at io.questdb.test.tools.TestUtils.assertMemoryLeak(TestUtils.java:461)
2022-06-06T19:21:23.2626996Z 	at io.questdb.cairo.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:381)
2022-06-06T19:21:23.2627760Z 	at io.questdb.cairo.AbstractCairoTest.assertMemoryLeak(AbstractCairoTest.java:376)
2022-06-06T19:21:23.2628550Z 	at io.questdb.cutlass.pgwire.PGJobContextTest.testUpdateAsync(PGJobContextTest.java:6139)
2022-06-06T19:21:23.2629361Z 	at io.questdb.cutlass.pgwire.PGJobContextTest.testUpdateAsync(PGJobContextTest.java:6107)
2022-06-06T19:21:23.2630076Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-06-06T19:21:23.2630780Z 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-06-06T19:21:23.2631599Z 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-06-06T19:21:23.2632318Z 	at java.lang.reflect.Method.invoke(Method.java:566)
2022-06-06T19:21:23.2633001Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-06-06T19:21:23.2633784Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-06-06T19:21:23.2634561Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-06-06T19:21:23.2635339Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-06-06T19:21:23.2636092Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-06-06T19:21:23.2636841Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2022-06-06T19:21:23.2637535Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2022-06-06T19:21:23.2638499Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
2022-06-06T19:21:23.2639370Z 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
2022-06-06T19:21:23.2640333Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-06-06T19:21:23.2640922Z 	at java.lang.Thread.run(Thread.java:829)
```

[62.txt.zip](https://github.com/questdb/questdb/files/8852695/62.txt.zip)

### Environment

```markdown
- **QuestDB version**:
- **OS**: 6.4.1-SNAPSHOT
- **Browser**:
```

**Comment by puzpuzpuz** [comment]
Maybe it be the same thing as what's addressed here https://github.com/questdb/questdb/blob/0c3c396b4f58609bc18ef1097d1edb40d4825e6a/core/src/test/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactoryTest.java#L498-L510 ?

**Comment by puzpuzpuz** [comment]
Should be fixed with #2263. Feel free to reopen if this test failure reoccurs.
